language/move-compiler/src/hlir/translate.rs

```diff
@@ -962,12 +953,30 @@ fn exp_<'env>(
                 let tbool = N::Type_::bool(loc);
                 let tu64 = N::Type_::u64(loc);
                 let tunit = sp(loc, N::Type_::Unit);
-                let vcond = Var(sp(loc, new_temp_name(stack.context)));
-                let vcode = Var(sp(loc, new_temp_name(stack.context)));
+                // name guaranteed to be unique, so we can skip the depth/color
+                let vcond_ = N::Var_ {
+                    name: new_temp_name(stack.context),
```

Comment:
Hmm, okay, this pattern might make it more difficult to implement the suggestion of a single count for both depth and colour, shared across all variable names...

Reply:
Although I notice that the temp name is derived from a counter in the context -- this could start where the proposed count from naming leaves off.

Reply:
Yeah that would definitely work too. I'm not sure either really feels simpler to me conceptually at first glance